### PR TITLE
Basic tests for Group

### PR DESCRIPTION
**Pull request by v-nikulichev**
This contains three basic tests for `Group`:
- **Ctor**. Make sure that constructor initializes all members.
- **ParseMetadataV1**. Generate msgpack buffer and check if it is parsed correctly. Metadata of version 1 contains array of integers -- group ids in a couple.
- **ParseMetadataV2**. The same as for V1, but parsing metadata version 2 format (dictionary).

### CODE REVIEW

File: tests/collector/group/group.cpp [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    packer.pack(std::string("status"));
+    packer.pack(std::string("MIGRATING"));
+    packer.pack(std::string(std::string("job_id")));
+    packer.pack(std::string("12345"));
+
+    // Create group and parse metadata.
+
+    Group g(29);
+    g.save_metadata(buffer.data(), buffer.size(), ::time(nullptr)*1000000000ULL);
+    ASSERT_EQ(0, g.parse_metadata());
```
Comment (shaitan) [inline tests/collector/group/group.cpp]: Why does `parse_metadata()` return `int`?

File: tests/collector/group/group.cpp [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+    // Check metadata values.
+
+    const Group::Metadata & md = g.get_metadata();
+    EXPECT_EQ(1, md.version);
+    EXPECT_FALSE(md.frozen);
+    EXPECT_EQ(std::vector<int>({17, 19, 23}), md.couple);
+    EXPECT_EQ("default", md.namespace_name);
+    EXPECT_TRUE(md.type.empty());
+    EXPECT_EQ(0, md.service.migrating);
```
Comment (shaitan) [inline tests/collector/group/group.cpp]: Why does it compare `migrating` with `0`?

File: tests/collector/group/group.cpp [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    packer.pack(17);
+    packer.pack(19);
+    packer.pack(23);
+
+    // Create group and parse metadata.
+
+    Group g(17);
+    g.save_metadata(buffer.data(), buffer.size(), ::time(nullptr)*1000000000ULL);
+    ASSERT_EQ(0, g.parse_metadata());
+    EXPECT_TRUE(g.metadata_parsed());
```
Comment (shaitan) [inline tests/collector/group/group.cpp:79]: It doesn't check `g.has_active_job()`.

**Comment by shaitan** [comment]
This test covers only positive scenarios, should it also include negative ones?

File: tests/collector/group/group.cpp [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    EXPECT_EQ(0, md.service.migrating);
+    EXPECT_TRUE(md.service.job_id.empty());
+
+    EXPECT_FALSE(g.metadata_parsed());
+    EXPECT_EQ(0, g.get_metadata_parse_duration());
+
+    EXPECT_EQ(Group::DATA, g.get_type());
+    EXPECT_EQ(Group::INIT, g.get_status());
+
+    EXPECT_FALSE(g.have_metadata_conflict(g));
```
Comment (shaitan) [inline tests/collector/group/group.cpp]: How can it be true?

File: tests/collector/group/group.cpp (hunk 3 above)
Comment (v-nikulichev) [inline tests/collector/group/group.cpp:79]: It is checked in Ctor test. Metadata parsing doesn't affect active_job.

Comment (shaitan) [inline tests/collector/group/group.cpp:79]: Should it be checked that it doesn't?

File: tests/collector/group/group.cpp (hunk 2 above)
Comment (v-nikulichev) [inline tests/collector/group/group.cpp]: Google Test gives warnings when I compare it with `false`. I made a workaround.

```
In file included from /home/saturnus/mastermind/tests/collector/googletest/googletest/include/gtest/gtest.h:1878:0,
                 from /home/saturnus/mastermind/tests/collector/group/group.cpp:22:
/home/saturnus/mastermind/tests/collector/group/group.cpp: In member function ‘virtual void Group_ParseMetadataV1_Test::TestBody()’:
/home/saturnus/mastermind/tests/collector/googletest/googletest/include/gtest/internal/gtest-internal.h:133:55: warning: converting ‘false’ to pointer type for argument 1 of ‘char testing::internal::IsNullLiteralHelper(testing::internal::Secret*)’ [-Wconversion-null]
     (sizeof(::testing::internal::IsNullLiteralHelper(x)) == 1)
                                                       ^
/home/saturnus/mastermind/tests/collector/googletest/googletest/include/gtest/gtest_pred_impl.h:77:52: note: in definition of macro ‘GTEST_ASSERT_’
   if (const ::testing::AssertionResult gtest_ar = (expression)) \
                                                    ^
/home/saturnus/mastermind/tests/collector/googletest/googletest/include/gtest/gtest_pred_impl.h:162:3: note: in expansion of macro ‘GTEST_PRED_FORMAT2_’
   GTEST_PRED_FORMAT2_(pred_format, v1, v2, GTEST_NONFATAL_FAILURE_)
   ^
/home/saturnus/mastermind/tests/collector/googletest/googletest/include/gtest/gtest.h:1927:3: note: in expansion of macro ‘EXPECT_PRED_FORMAT2’
   EXPECT_PRED_FORMAT2(::testing::internal:: \
   ^
/home/saturnus/mastermind/tests/collector/googletest/googletest/include/gtest/gtest.h:1928:32: note: in expansion of macro ‘GTEST_IS_NULL_LITERAL_’
                       EqHelper<GTEST_IS_NULL_LITERAL_(expected)>::Compare, \
                                ^
/home/saturnus/mastermind/tests/collector/group/group.cpp:91:5: note: in expansion of macro ‘EXPECT_EQ’
     EXPECT_EQ(false, md.service.migrating);
     ^
```

File: tests/collector/group/group.cpp (hunk 1 above)
Comment (v-nikulichev) [inline tests/collector/group/group.cpp]: Changed to `bool`.
https://github.com/v-nikulichev/mastermind/commit/2a852a38bbc9090a134dd23a4508b10ab85bd2dd
https://github.com/v-nikulichev/mastermind/commit/fbb4eaa191d8e21ad504147b7a24d6f0fae2d10b

File: tests/collector/group/group.cpp (hunk 2 above)
Comment (v-nikulichev) [inline tests/collector/group/group.cpp]: Now using EXPECT_FALSE: https://github.com/v-nikulichev/mastermind/commit/41ccb5cefee2ec16dcf3abef9f778c8d07091005

File: tests/collector/group/group.cpp (hunk 4 above)
Comment (v-nikulichev) [inline tests/collector/group/group.cpp]: Just simple sanity check.

Comment (shaitan) [inline tests/collector/group/group.cpp]: What does it check? It is like:

``` c++
int a = 1;
EXPECT_FALSE(a!=a);
```

**Pull request merged**